src/upgrades/mongodb_upgrade.py

```diff
@@ -419,6 +558,82 @@ def step_down_primary_and_wait_reelection(self) -> None:
                 if new_primary == old_primary:
                     raise FailedToElectNewPrimaryError()
 
+    def are_pre_upgrade_operations_config_server_successful(self):
+        """Runs pre-upgrade operations for config-server and returns True if successful."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            return False
+
+        if not self.is_feature_compatibility_version(Config.Upgrade.FEATURE_VERSION_6):
+            logger.debug(
+                "Not all replicas have the expected feature compatibility: %s",
+                Config.Upgrade.FEATURE_VERSION_6,
+            )
+            return False
+
+        self.set_mongos_feature_compatibilty_version(Config.Upgrade.FEATURE_VERSION_6)
+
+        try:
+            self.turn_off_and_wait_for_balancer()
```

Comment:
Related to the previous comment - will this run for every unit upgrading?

Reply:
No. the pre-upgrade check runs:
- once on the leader unit (from `juju run pre-upgrade-check`)
- once on the highest unit (from `juju refresh <charm> <options>`)